File: src/main/java/ca/joyfactory/coims/domain/CaseTypeFee.java
package ca.joyfactory.coims.domain;

import lombok.Data;

import javax.persistence.*;

/**
 * Created by Joinsu on 2018-08-31.
 */
@Entity
public class CaseTypeFee {

    @Id
    @Column(name = "FEE_ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(length = 10)
    private double fee = 0.00;

    @ManyToOne( cascade = CascadeType.PERSIST)
    @JoinColumn( name = "COMPANY_ID", nullable = false)
    private Company company;

    @Column(length = 10, nullable = false)
    private Long caseTypeId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public double getFee() {
        return fee;
    }

    public void setFee(double fee) {
        this.fee = fee;
    }

    public Company getCompany() {
        return company;
    }

    public void setCompany(Company company) {
        if( this.company != null){
            this.company.getCaseTypeFees().remove( this);
        }

        this.company = company;

        if( !this.company.getCaseTypeFees().contains( this)){
            this.company.getCaseTypeFees().add( this);
        }
    }

    public Long getCaseTypeId() {
        return caseTypeId;
    }

    public void setCaseTypeId(Long caseTypeId) {
        this.caseTypeId = caseTypeId;
    }
}

File: frontend/src/app/_model/case-type-fee.ts
/**
 * Created by Joinsu on 2018-11-23.
 */
export class CaseTypeFee{
  private id: number;
  private fee: number;
  private caseTypeId: number;

  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }

  getFee(): number {
    return this.fee;
  }

  setFee(value: number) {
    this.fee = value;
  }

  getCaseTypeId(): number {
    return this.caseTypeId;
  }

  setCaseTypeId(value: number) {
    this.caseTypeId = value;
  }

  setDataFromResponse( responseData: any){
    this.setId( responseData.id);
    this.setFee( responseData.fee);
    this.setCaseTypeId( responseData.caseTypeId);
  }
}
